.github/workflows/check-measurements-reproducibility.yml

```
@@ -0,0 +1,123 @@
+name: Check measurements reproducibility
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        description: The version of the measurements that are downloaded from the CDN.
+        required: true
+      ref:
+        type: string
+        description: The git ref to check out. You probably want this to be the tag of the release you are testing.
+        required: true
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        description: The version of the measurements that are downloaded from the CDN.
+        required: true
+      ref:
+        type: string
+        description: The git ref to check out.
+        required: true
+
+jobs:
+  check-reproducibility:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ github.event.inputs.ref }}
+      - name: Set up bazel
+        uses: ./.github/actions/setup_bazel_nix
+        with:
+          useCache: "false"
+          nixTools: |
+            systemdUkify
+            jq
+            jd-diff-patch
+            moreutils
+      - name: Build images
+        id: build-images
+        run: |
+          set -euo pipefail
+
+          # Build required binaries
+          bazel build //image/system:stable
+          bazel build //image/measured-boot/cmd
+          echo "buildPath=$PWD/bazel-bin/image" | tee -a "$GITHUB_OUTPUT"
+          cd "$(mktemp -d)"
+
+      - name: Download measurements
+        run: |
+          curl -O https://cdn.confidential.cloud/constellation/v2/ref/-/stream/stable/${{ github.event.inputs.version }}/image/measurements.json
+      
+      - name: Cleanup release measurements and generate our own
+        run: |
+          set -euo pipefail
+          shopt -s extglob
+
+          for directory in ${{ steps.build-images.outputs.buildPath }}/system/!(mkosi_wrapper.sh); do
+            dirname="$(basename "$directory")"
+            csp="$(echo "$dirname" | cut -d_ -f1)"
+            attestationVariant="$(echo "$dirname" | cut -d_ -f2)"
+
+            echo "Comparing measurements of CSP $csp with attestation variant $attestationVariant"
+            # This jq filter selects the measurements for the correct CSP and attestation variant
+            # and then removes all `warnOnly: true` measurements.
+            jq --arg attestation_variant "$attestationVariant" --arg csp "$csp" \
+              '
+                .list.[]
+                | select(
+                  .attestationVariant == $attestation_variant
+                )
+                | select((.csp | ascii_downcase) == $csp)
+                | .measurements
+                | walk(
+                    if (
+                      type=="object" and .warnOnly
+                    )
+                    then del(.) else . end
+                  )
+                | del(..|nulls)
```

```suggestion
                | to_entries | map(select(.value.warnOnly | not)) | from_entries
```
I would prefer keeping this functional/declarative, as that is the prevailing style of `jq` programs.